File: android/core/data/src/main/kotlin/com/aibook/android/core/data/db/OpdsCatalogEntryEntity.kt
package com.aibook.android.core.data.db

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "opds_catalog_entries",
    indices = [
        Index(value = ["connectionId"]),
        Index(value = ["connectionId", "acquisitionHref"], unique = true)
    ]
)
data class OpdsCatalogEntryEntity(
    @PrimaryKey
    val id: String,
    val connectionId: String,
    val sourceName: String,
    val title: String,
    val author: String? = null,
    val summary: String? = null,
    val coverHref: String? = null,
    val acquisitionHref: String,
    val acquisitionType: String? = null,
    val format: String,
    val categories: String = "",
    val syncedAt: Long
)

File: ios/AiBook/Core/Data/Database/OpdsCatalogEntryEntity.swift
import Foundation
import SwiftData

// MARK: - OpdsCatalogEntryEntity（与安卓 OpdsCatalogEntryEntity 对齐）

@Model
final class OpdsCatalogEntryEntity {
    var id: String = UUID().uuidString
    var connectionId: String = ""
    var title: String = ""
    var author: String = ""
    var summary: String = ""
    var acquisitionHref: String = ""
    var acquisitionType: String = ""
    var coverUri: String = ""
    var syncedAt: Date = Date()

    init() {}

    func toDisplay() -> OpdsCatalogEntryDisplay {
        OpdsCatalogEntryDisplay(
            title: title,
            author: author.isEmpty ? nil : author,
            summary: summary.isEmpty ? nil : summary,
            acquisitionHref: acquisitionHref.isEmpty ? nil : acquisitionHref,
            acquisitionType: acquisitionType.isEmpty ? nil : acquisitionType,
            coverUri: coverUri.isEmpty ? nil : coverUri
        )
    }
}
